SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -949,3 +952,124 @@ void USpatialSender::AddTombstoneToEntity(const Worker_EntityId EntityId)
+
+	// Process InitialOnly data of actor
+	FActorSchemaData& ActorSchemaData = SchemaDataBase->ActorClassPathToSchema[ClassPathName];
+	int32_t InitialOnlyComponentId = ActorSchemaData.SchemaComponents[SCHEMA_InitialOnly];
+	if (InitialOnlyComponentId && InitialOnlyComponentId != SCHEMA_Invalid)
+	{
+		OutInitialOnlyComponentIds.Add(InitialOnlyComponentId);
+	}
+
+	// Process InitialOnly data of subobject
+	for (auto SubobjectSchemaDataKV : ActorSchemaData.SubobjectData)
+	{
+		int32_t TmpInitialOnlyComponentId = SubobjectSchemaDataKV.Value.SchemaComponents[SCHEMA_InitialOnly];
+		if (TmpInitialOnlyComponentId && TmpInitialOnlyComponentId != SCHEMA_Invalid)
+		{
+			OutInitialOnlyComponentIds.Add(TmpInitialOnlyComponentId);
+		}
+	}
+
+	// Process InitialOnly data of dynamic subobject
+	for (auto ActorComponent : ActorComponents)
+	{
+		FString TmpClassPathName = ActorComponent->GetClass()->GetPathName();
+		if (SchemaDataBase->SubobjectClassPathToSchema.Contains(TmpClassPathName))
+		{
+			FSubobjectSchemaData& SubobjectSchemaData = SchemaDataBase->SubobjectClassPathToSchema[TmpClassPathName];
+			for (auto DynamicSubobjectComponents : SubobjectSchemaData.DynamicSubobjectComponents)
+			{
+				int32_t TmpInitialOnlyComponentId = DynamicSubobjectComponents.SchemaComponents[SCHEMA_InitialOnly];
+				if (TmpInitialOnlyComponentId && TmpInitialOnlyComponentId != SCHEMA_Invalid)
+				{
+					OutInitialOnlyComponentIds.Add(TmpInitialOnlyComponentId);
+				}
+			}
+		}
+	}
+}
+
+void USpatialSender::SendInitialOnlyRequest(Worker_EntityId EntityId)
+{
+	TArray<Worker_ComponentId> LocalInitialOnlyComponentIds;
+	GetInitialOnlyComponentIds(EntityId, LocalInitialOnlyComponentIds);
+
+	if (0 == LocalInitialOnlyComponentIds.Num())
+	{
+		return;
+	}
+
+	InitialOnlyRequests.Add(TPair<Worker_EntityId, TSet<Worker_ComponentId>>(EntityId, LocalInitialOnlyComponentIds));
+}
+
+void USpatialSender::CancelInitialOnlyRequest(Worker_EntityId EntityId)
+{
+	InitialOnlyRequests.Remove(EntityId);
+}
+
+void USpatialSender::FlushInitialOnlyRequests()
+{
+	if (0 == InitialOnlyRequests.Num())
+	{
+		return;
+	}
+
+	int32_t EntityCount = 0;
+
+	TSet<Worker_ComponentId> ComponentIdsSet;
+	TArray<Worker_Constraint> EntityConstraintArray;
+	TArray<Worker_EntityId> RemovedEntities;
+
+	for (auto Entity : InitialOnlyRequests)
+	{
+		Worker_EntityId TmpEntityId = Entity.Key;
+		Worker_Constraint Constraints;
+		Constraints.constraint_type = WORKER_CONSTRAINT_TYPE_ENTITY_ID;
+		Constraints.constraint.entity_id_constraint.entity_id = TmpEntityId;
+
+		EntityConstraintArray.Add(Constraints);
+		ComponentIdsSet.Append(Entity.Value);
+		RemovedEntities.Add(Entity.Key);
+
+		if (++EntityCount >= MaxInitialOnlyRequestPerTick)
+		{
+			break;
+		}
+	}
+
+	for (auto RemovedEntityId : RemovedEntities)
+	{
+		InitialOnlyRequests.Remove(RemovedEntityId);
+	}
+
+	Worker_Constraint InitialOnlyConstraints;
+	InitialOnlyConstraints.constraint_type = WORKER_CONSTRAINT_TYPE_OR;
```

Comment:
Using the OR type but the AND object seems like a mismatch - I'm not sure how this functions

Reply:
You are right, this is a mistake, should be `or_constraint` object instead